### PR TITLE
fix: make host param compulsory

### PR DESCRIPTION
**Pull request by ahtesham-quraish**
Description
Make host param compulsory and host should grab from query params when calling postMessage api

its related to : [VAN-1485](https://2u-internal.atlassian.net/browse/VAN-1485)

its tested locally.

### CODE REVIEW

File: src/common-components/RedirectLogistration.jsx [hunk 1]
```diff
@@ -88,6 +89,7 @@ RedirectLogistration.defaultProps = {
   redirectToRecommendationsPage: false,
   userId: null,
   registrationEmbedded: false,
+  host: null,
```
Comment (zainab-amir) [inline src/common-components/RedirectLogistration.jsx:92]:
```suggestion
  host:  '',
```

File: src/data/utils/dataUtils.js [hunk 2]
```diff
@@ -79,5 +79,5 @@ export const windowScrollTo = (options) => {
 
 export const isRegistrationEmbedded = () => {
   const queryParams = getAllPossibleQueryParams();
-  return queryParams?.variant === EMBEDDED;
+  return queryParams?.variant === EMBEDDED && 'host' in queryParams;;
```
Comment (zainab-amir) [inline src/data/utils/dataUtils.js]:
```suggestion
  return queryParams?.variant === EMBEDDED && 'host' in queryParams;
```

File: src/common-components/RedirectLogistration.jsx [hunk 3]
```diff
@@ -40,7 +41,7 @@ const RedirectLogistration = (props) => {
       setCookie('van-504-returning-user', true);
 
       if (registrationEmbedded) {
-        window.parent.postMessage({ action: REDIRECT, redirectUrl: `${window.location.origin}${AUTHN_PROGRESSIVE_PROFILING}` }, DISCOVER_URL);
+        window.parent.postMessage({ action: REDIRECT, redirectUrl: `${window.location.origin}${AUTHN_PROGRESSIVE_PROFILING}` }, host);
```
Comment (zainab-amir) [inline src/common-components/RedirectLogistration.jsx:44]: Add the url in the edx-internal config
```suggestion
        window.parent.postMessage(
            {
                action: REDIRECT,
                redirectUrl: getConfig().POST_RESISTRATION_REDIRECT_URL
             },
            host
         );
```

File: src/common-components/tests/UnAuthOnlyRoute.test.jsx [hunk 4]
```diff
@@ -96,7 +96,7 @@ describe('UnAuthOnlyRoute', () => {
     };
 
     delete window.location;
-    window.location = { href: getConfig().BASE_URL.concat(REGISTER_PAGE), search: '?variant=embedded' };
+    window.location = { href: getConfig().BASE_URL.concat(REGISTER_PAGE), search: `?variant=embedded&host=${DISCOVER_URL}` };
```
Comment (zainab-amir) [inline src/common-components/tests/UnAuthOnlyRoute.test.jsx:99]: Let's not use discover social url in tests and we don't really need a constant. Just use any dummy url i.e
```suggestion
    window.location = { href: getConfig().BASE_URL.concat(REGISTER_PAGE), search: '?variant=embedded&host=http://localhost/host-website' };
```

**Review by zainab-amir — CHANGES_REQUESTED** [review]

File: .env [hunk 5]
```diff
@@ -19,6 +19,7 @@ REGISTER_CONVERSION_COOKIE_NAME=null
 # ***** Links *****
 LOGIN_ISSUE_SUPPORT_LINK=''
 AUTHN_PROGRESSIVE_PROFILING_SUPPORT_LINK=null
+POST_RESISTRATION_REDIRECT_URL=''
```
Comment (zainab-amir) [inline .env:22]: typo:
```suggestion
POST_REGISTRATION_REDIRECT_URL=''
```

File: src/common-components/RedirectLogistration.jsx (hunk 3 above)
Comment (zainab-amir) [reply src/common-components/RedirectLogistration.jsx]: typo: POST_REGISTRATION_REDIRECT_URL

File: src/config/index.js [hunk 6]
```diff
@@ -14,6 +14,7 @@ const configuration = {
   AUTHN_PROGRESSIVE_PROFILING_SUPPORT_LINK: process.env.AUTHN_PROGRESSIVE_PROFILING_SUPPORT_LINK || null,
   LOGIN_ISSUE_SUPPORT_LINK: process.env.LOGIN_ISSUE_SUPPORT_LINK || null,
   PASSWORD_RESET_SUPPORT_LINK: process.env.PASSWORD_RESET_SUPPORT_LINK || null,
+  POST_REGISTRATION_REDIRECT_URL: process.env.POST_RESISTRATION_REDIRECT_URL || '',
```
Comment (syedsajjadkazmii) [inline src/config/index.js:17]:
```suggestion
  POST_REGISTRATION_REDIRECT_URL: process.env.POST_REGISTRATION_REDIRECT_URL || '',
```

**Review by zainab-amir — APPROVED** [review]

**Pull request merged**